AppCenterIdentity/AppCenterIdentity/MSIdentity.m

```diff
@@ -310,4 +305,60 @@ - (void)removeAuthToken {
   }
 }
 
+- (void)acquireTokenSilentlyWithMSALAccount:(MSALAccount *)account {
+  __weak typeof(self) weakSelf = self;
+  [self.clientApplication
+      acquireTokenSilentForScopes:@[ (NSString * _Nonnull) self.identityConfig.identityScope ]
+                          account:account
+                  completionBlock:^(MSALResult *result, NSError *e) {
+                    typeof(self) strongSelf = weakSelf;
+                    if (e) {
+                      MSLogWarning([MSIdentity logTag],
+                                   @"Silent acquisition of token failed with error: %@. Triggering interactive acquisition", e);
+                      [strongSelf acquireTokenInteractively];
+                    } else {
+                      [MSAuthTokenContext sharedInstance].authToken = result.idToken;
+                      [strongSelf saveAuthToken:result.idToken];
+                      [strongSelf saveAccountId:(NSString * _Nonnull) result.account.homeAccountId];
+                    }
+                  }];
+}
+
+- (void)acquireTokenInteractively {
+  __weak typeof(self) weakSelf = self;
+  [self.clientApplication acquireTokenForScopes:@[ (NSString * _Nonnull) self.identityConfig.identityScope ]
+                                completionBlock:^(MSALResult *result, NSError *e) {
+                                  if (e) {
+                                    MSLogError([MSIdentity logTag], @"User sign-in failed. Error: %@", e);
+                                  } else {
+                                    typeof(self) strongSelf = weakSelf;
+                                    [MSAuthTokenContext sharedInstance].authToken = result.idToken;
+                                    [strongSelf saveAuthToken:result.idToken];
+                                    [strongSelf saveAccountId:(NSString * _Nonnull) result.account.homeAccountId];
+                                  }
+                                }];
+}
+
+- (MSALAccount *)retrieveAccount {
```

Comment:
Please add a method similar to `saveAccountId` / `removeAccountId` for consistency (and I would also pass account id as a parameter here - due to single responsibility principle).
The account id getter is also going to be useful in Push implementation (we need to retrieve account id there).

Reply:
make sense, will do